src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -200,8 +200,8 @@ public OptionSet Options
             {
                 return this.CurrentSolution.Options;
             }
-            // TODO: mark the setter as deprecated, add a new API that clearly indicates to the caller that underlying CurrentSolution changed,
-            //       and migrate all the callsites of this setter to the new API.
+
+            [Obsolete(@"Workspace options should be set by invoking 'workspace.TryApplyChanges(workspace.CurrentSolution.WithOptions(newOptionSet))'")]
```

Comment:
❔ To be clear: this is a deprecation in favor of a superior API with no intent to ever actually stop supporting the current behavior, correct?

While in the future it may not work to use `solution.Workspace.Options = ...`, it will always be possible to set options on the root `Workspace` (e.g. a `new AdhocWorkspace()`) and then obtain the solution from it.

Reply:
> this is a deprecation in favor of a superior API with no intent to ever actually stop supporting the current behavior, correct?

Yes, that is my understanding.